src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs

```diff
@@ -4979,6 +4979,40 @@ struct A
                 Diagnostic(ErrorCode.ERR_AmbigUDConv, "b2").WithArguments("B.implicit operator A(B)", "A.implicit operator A(B)", "B", "A"));
         }
 
+        [WorkItem(545631, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545631")]
+        [Fact]
+        public void TestNullCoalesceWithInvalidUserDefinedConversions_04()
+        {
+            var source = @"
+struct B
+{
+    static void Main()
+    {
+        A a2;
+        B? b2 = null;
+        var c2 = b2 ?? a2;
```

Comment:
Consider making this just an assignment, rather than relying on the rules of `??` to probably do the thing you want.

Reply:
Ended up replacing this with a test in FlowTests which attempts more directly to get into a situation where `CanPropagateStateWhenNotNull` could be called with a user-defined conversion with an unexpected parameter count. The method is shared between definite assignment and nullable so I feel like the level of testing introduced in the latest commit is adequate.